tensornetwork/network.py

```diff
@@ -470,7 +470,10 @@ def outer_product(self,
     Returns:
       A new node. Its shape will be node1.shape + node2.shape
     """
-    new_tensor = self.backend.outer_product(node1.tensor, node2.tensor)
+    if node1.get_rank() == 0 or node2.get_rank() == 0:
+      new_tensor = node1.tensor * node2.tensor
```

Comment:
To help with supporting future backends (Things like the `jax_tpu` backend that is in development https://github.com/google/TensorNetwork/pull/193 and the `ShellBackend`), can you make this a new operation on the backend object? So something like `self.backend.multiply(...)`?